src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -10563,13 +10563,21 @@ private void prepare() {
     /**
      * Coerce an object instance to a boolean value.
      * An object is coerced to true if it's not null, to false if it is null.
+     * One exception to this is in the case of a non-null {@code java.util.Optional}
+     * in which case the result of the {@code isPresent} method is returned.
      *
      * @param object the object to coerce
      * @return the boolean value
      * @since 1.7.0
      */
     public static boolean asBoolean(Object object) {
-        return object != null;
+        if (object == null) {
+            return false;
+        }
+        if ("java.util.Optional".equals(object.getClass().getName())) {
```

Comment:
just a very small thing.. could you add a todo to move this into a jvm8 module once we decided to make that our minimum jdk for the build?